M3-4388 Responsive styling for Entity Header (All variants)

WilkinsKa1 opened the pull request:
## Description

This PR adjusts responsive styling for both variants of the entity header (landing and details). Please test various breakpoints thoroughly, and make use of chrome's device emulator and/or Xcode to get an idea of what things will look like for particular devices. I did test 'in-between' breakpoints and made fixes where I could, but in order to get everything working for both variants there are sure to be some less-than-optimal styling decisions I made that weren't included in the designs.

## Type of Change
- Non breaking change ('update')

## Note to Reviewers

Notes will be throughout to explain some of the changes!

WilkinsKa1 commented on packages/manager/src/MainContent_CMR.tsx:
```diff
@@ -59,8 +59,8 @@ const useStyles = makeStyles((theme: Theme) => ({
       paddingTop: theme.spacing(2),
-      paddingLeft: theme.spacing(2),
-      paddingRight: theme.spacing(2)
+      paddingLeft: 0,
```
Needed to remove this here so tablet views would be accurate. This is technically part of the work for the main content width.

WilkinsKa1 commented on packages/manager/src/components/EntityHeader/EntityHeader.tsx:
```diff
@@ -37,6 +49,10 @@ const useStyles = makeStyles((theme: Theme) => ({
       letterSpacing: '.5px'
     }
+  },
+  bodyDetailVariant: {
```
This is for the details variant as the name implies- still also inheriting from contentOuter (the OG class) that landing has as well.

WilkinsKa1 commented on packages/manager/src/components/TagCell/TagCell.tsx:
```diff
@@ -13,6 +13,12 @@ const useStyles = makeStyles((theme: Theme) => ({
   root: {
     position: 'relative'
   },
+  rootDetails: {
```
More conditional styling based on context within the app- we're doing quite a bit of this for CMR.

Jskobos commented on packages/manager/src/components/EntityHeader/EntityHeader.tsx:
```diff
@@ -53,20 +69,23 @@ export const EntityHeader: React.FC<HeaderProps> = props => {
+          parentText={parentText}
+        />
+
+        <Hidden smDown>
```
This is causing /domains to crash, bc you're not allowed to have a render function that doesn't render; to render nothing, you have to explicitly return `null`. Has to be `body ? <Grid ..../> : null`

WilkinsKa1 commented:
Thanks for the feedback @Jskobos ! I addressed the issues you called out- I decided to wrap the clear filters button on smaller screens, lmk what you think.

WilkinsKa1 commented:
Thanks for the reviews, @johnwcallahan @tiffwong ! I’ve addressed the following:

• Domains landing
• wrapping for in-between breakpoints
• IP Table header
• Dark mode divider
• Clear filters alignment

Good question in re breakpoints! The lowest I look at is iPhone 6/7/8 plus (this goes for current CM as well).

johnwcallahan commented:
Thanks @WilkinsKa1, adjustments look good. One thing I didn't notice before: in grid view (landing variant) components aren't collapsing the way they are on the detail pages:
![Screen Shot 2020-07-29 at 8 47 45 AM](https://user-images.githubusercontent.com/16911484/88802078-91c35600-d178-11ea-8086-2b70c35cfee1.png)

WilkinsKa1 commented:
@johnwcallahan thanks for checking that- I didn't even realize that wasn't quite the details variant. Made adjustments to account for that as well!

WilkinsKa1 commented on packages/manager/src/components/EntityHeader/EntityHeader.tsx:
```diff
@@ -3,32 +3,57 @@ import * as classnames from 'classnames';
   title: string | JSX.Element;
   bodyClassName?: string;
   isLanding?: boolean;
+  isSecondary?: boolean;
```
Note 2 'minor' variants here- one is for 'secondary' landings, such as Ip Addresses table header, and the isDetailsLanding is for card view, which is actually a variation of the true details view as it does not have a breadcrumb and has additional elements we needed to account for.
